### PR TITLE
LungCancerDetection initial commit

### PR DESCRIPTION
**Pull request by tyler-romero**
Lung Cancer Detection extended example.

### CODE REVIEW

**Comment by tyler-romero** [comment]
@Xue001 @saroufimc1 @sdgilley
No option to add reviews without write permission, so I'm tagging you in the comments.

File: LungCancerDetection/INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+You need to change the user and password in SQL Server. For that you have to execute the file `change_auth.sql` (changing the variables user and password).
+
+#### Install the rest of the libraries needed to run the the demo
+The next step is to install several libraries that we need to run the demo. First you need to install OpenCV:
+
+	cd "C:\Program Files\Microsoft SQL Server\YOUR-MSSQL-SERVER-INSTANCE-FOLDER\PYTHON_SERVICES\Scripts"
+	conda.exe install -c https://conda.binstar.org/conda-forge opencv -y
+
+Finally, you need to install the libraries available in the `requirements.txt`. For that you have execute SQL `pip` inside the directory where you downloaded this repo. So, in a terminal write:
```
Comment (tritm78) [inline LungCancerDetection/INSTALL.md]: add link to the file location?

File: LungCancerDetection/INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Finally, to test that everything works correctly you have to execute the file [sqlpytest.sql](sql/sqlpytest.sql).
+
+You need to change the user and password in SQL Server. For that you have to execute the file `change_auth.sql` (changing the variables user and password).
+
+#### Install the rest of the libraries needed to run the the demo
+The next step is to install several libraries that we need to run the demo. First you need to install OpenCV:
+
+	cd "C:\Program Files\Microsoft SQL Server\YOUR-MSSQL-SERVER-INSTANCE-FOLDER\PYTHON_SERVICES\Scripts"
+	conda.exe install -c https://conda.binstar.org/conda-forge opencv -y
```
Comment (tritm78) [inline LungCancerDetection/INSTALL.md]: I actually ran into the following error when running this line, and found a solution by setting the environment variable CONDA_SSL_VERIFY=false from this link (https://github.com/conda/conda/issues/5170):
Fetching package metadata ...ERROR urllib3.connection:_match_hostname(360): Cert
ificate did not match expected hostname: conda.binstar.org. Certificate: {'subje
ct': ((('commonName', 'anaconda.com'),),), 'subjectAltName': [('DNS', 'anaconda.
com'), ('DNS', 'anacondacloud.com'), ('DNS', 'anacondacloud.org'), ('DNS', 'bins
tar.org'), ('DNS', 'wakari.io')]}

File: LungCancerDetection/INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Finally, you need to install the libraries available in the `requirements.txt`. For that you have execute SQL `pip` inside the directory where you downloaded this repo. So, in a terminal write:
+
+	"C:\Program Files\Microsoft SQL Server\YOUR-MSSQL-SERVER-INSTANCE-FOLDER\PYTHON_SERVICES\Scripts\pip.exe" install -r requirements.txt
+
+#### Install lung cancer detection libraries
+The last step is to install the lung cancer libraries. You have to go to the folder where you downloaded the libraries and fill up the file [connection_settings.py](lung_cancer/connection_settings.py.template) with your credentials. Another way to add the credentials is using environmnet variables. In this case we use a file embedded in the library because it's easier with the SQL python system. Once the credentials are changed, execute from there:
```
Comment (tritm78) [inline LungCancerDetection/INSTALL.md]: typo 'environmnet'?

File: LungCancerDetection/README.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	<img src="https://msdnshared.blob.core.windows.net/media/2017/02/021717_1842_QuickStartG3.png" alt="lung cancer scans" width="30%"/>
+</p>
+
+We are going to upload the images to SQL. The reason for doing this, instead of reading the images directly from disk, is because we want to simulate an scenario where all the data is already in SQL. For demo purposes we are going to use a small subset of the images, they can be found in [stage1_labels_partial.csv](data/stage1_labels_partial.csv). This subset consists of 200 patients out of 1595. The complete patient info is [stage1_labels.csv](data/stage1_labels.csv).
+
+The first step is to create in SQL Server a database called `LungCancer`. We also require that FileTables are enabled on the database. This is for working with image data in SQL Server. Follow the instructions [here](https://docs.microsoft.com/en-us/sql/relational-databases/blob/enable-the-prerequisites-for-filetable) to set it up. Execute [file_table_creation.sql](sql/file_table_creation.sql) in order to create this database and allow it to support FileTables. This script will also create a FileTable called `MriData`.
```
Comment (tritm78) [inline LungCancerDetection/README.md]: After the database is created, in SSMS, right-click on the database and go to properties and then FileGroups to add a new filegroup, and then go to Files to create a database file.

File: LungCancerDetection/README.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+In addition, SQLRUsergroup user group should be granted dbreader access on your database. Otherwise, the sp_execute_external_script won't be able to access the image files.
```
Comment (tritm78) [inline LungCancerDetection/README.md]: need to clarify how to do it?

File: LungCancerDetection/README.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+The next step is to upload the images. First you need to put the correct paths in the file [config_preprocessing.py](preprocessing/config_preprocessing.py.template). In case you want to upload the full dataset, just uncomment `STAGE1_LABELS = os.path.join(DATA_PATH, 'stage1_labels.csv')`. In addition, you need to set `IMAGES_FOLDER` to the location of your file table. To import the images to the SQL database using the FileTable you have set up, you have to execute the script [insert_scan_images_in_sql_database.py](preprocessing/insert_scan_images_in_sql_database.py). This will take a while.
```
Comment (tritm78) [inline LungCancerDetection/README.md]: not clear how to set IMAGES_FOLDER and STAGE1_FOLDER?

Comment (tritm78) [reply LungCancerDetection/README.md]: IMAGES_FOLDER should be set in connection_strings.py, something like: IMAGES_FOLDER =  '\\\\Tyler-Laptop\\tylersqlserver\\DemoData\\MriData' To file the path, right-click on the file table (dbo.MriData) and select Explore FileTable Directory

Comment (tritm78) [reply LungCancerDetection/README.md]: STAGE1_FOLDER should be set in config_preprocessing.py and point to where the downloaded dataset has been extracted to.

File: LungCancerDetection/INSTALL.md (hunk 3 above)
Comment (tyler-romero) [reply LungCancerDetection/INSTALL.md]: Fixed

File: LungCancerDetection/README.md (hunk 4 above)
Comment (tyler-romero) [reply LungCancerDetection/README.md]: Added

File: LungCancerDetection/README.md (hunk 5 above)
Comment (tyler-romero) [reply LungCancerDetection/README.md]: Done.

File: LungCancerDetection/README.md (hunk 6 above)
Comment (tyler-romero) [reply LungCancerDetection/README.md]: Updated

File: LungCancerDetection/INSTALL.md (hunk 1 above)
Comment (tyler-romero) [reply LungCancerDetection/INSTALL.md]: Done

File: LungCancerDetection/INSTALL.md (hunk 2 above)
Comment (tyler-romero) [reply LungCancerDetection/INSTALL.md]: Fixed

**Comment by jeroenterheerdt** [comment]
@tyler-romero I wanted to go ahead and merge your PR (from a while back, sorry). I cannot however, since you have not signed the ['Contributed License Agreement'](https://cla.opensource.microsoft.com/Microsoft/SQL-Server-R-Services-Samples?pullRequest=27). Can you please sign it? Otherwise I will not be able to merge your excellent sample into the repo.

**Pull request closed without merging**